gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -91,10 +101,22 @@ public void onViewCreated(
     viewModel.getForm().observe(this, this::rebuildForm);
     viewModel.getToolbarTitle().observe(this, toolbar::setTitle);
     viewModel.getSaveResults().observe(this, e -> e.ifUnhandled(this::handleSaveResult));
+    viewModel.getAddedPhoto().observe(this, this::onPhotoAdded);
     // Initialize view model.
     viewModel.initialize(EditObservationFragmentArgs.fromBundle(getArguments()));
   }
 
+  private void onPhotoAdded(Map<Field, File> fieldFileMap) {
+    Log.d(TAG, fieldFileMap.toString());
+
+    // TODO: Upload photo to Firestore Storage
+    // TODO: Fetch download url and update response in viewModel
+    for (Field field : fieldFileMap.keySet()) {
+      File file = fieldFileMap.get(field);
+      viewModel.onResponseChanged(field, TextResponse.fromString(file.getPath()));
```

Comment:
What does this do, and why does it do it for all image fields, not just the changed one?

Reply:
Cleaned up